src/g2grids.f

```diff
@@ -118,12 +118,12 @@ integer function readgrids(lunit)
              gtemp%gridtmpl=igdtmpl
              gtemp%cdesc=desc
              nullify(gtemp%next)              ! defines end of linked list.
+             prev => gtemp
              if ( count .eq. 1 ) then
                 gridlist => gtemp
              else                       ! make sure previous entry in list
                 prev%next => gtemp      ! points to the new entry,
              endif
-             prev => gtemp
```

Comment:
I'm not sure this is what's wanted. How will we set prev%next properly?

Reply:
Maybe move `prev => gtemp` inside the if statement before else, then also add it after `prev%next => gtemp` ?